pkg/urlutil/urlerrorhelper.go

```diff
@@ -0,0 +1,26 @@
+package urlutil
+
+import (
+	"fmt"
+	"net/url"
+	"strings"
+)
+
+func RemoveUrlFromErr(err error) error {
+	strSegments := strings.Split(err.Error(), " ")
+	for i, v := range strSegments {
+		if IsValidUrl(v) {
+			// we found a url
+			strSegments[i] = "[REDACTED]"
+		}
+	}
+	return fmt.Errorf(strings.Join(strSegments, " "))
+}
+
+func IsValidUrl(urlstring string) bool {
+	u, parseError := url.Parse(urlstring)
+	if parseError == nil && u.Scheme != "" && u.Host != "" && u.Path != "" {
```

Comment:
what if there's no path? - https://agent.fsb.ru/?sekret=12345

Reply:
Good point, I had overlooked that possibility.